home-work3/utils.js

```
@@ -0,0 +1,113 @@
+                    data.push(row);
+                }
+            })
+
+            .on('end', () => {
+                resolve(data)
+            })
+    }))
+};
+
+const getEventById = async (id) => {
+    return new Promise(((resolve, reject) => {
+        let data;
+        fs.createReadStream(filePath)
+            .on('error', (err) => {
+                reject(err)
+            })
+
+            .pipe(csv())
+            .on('data', (row) => {
+                if (id.toString() === row.Id.toString()) {
+                    data = row;
+                    fs.createReadStream(filePath).close();
+                }
+            })
+
+            .on('end', () => {
+                resolve(data);
+            })
+    }))
+};
+
+const addEvent = async event => {
+    const {Id, Title, Location, Date, Hour} = event;
+    return new Promise((resolve, reject) => {
+        fs.createWriteStream(filePath, {flags: 'a'})
+            .on('error', (err) => {
+                reject(err);
+            })
+            .on('finish', () => {
+                resolve(event);
+            })
+            .write(`${Id},${Title},${Location},${Date},${Hour} \n`);
+    });
+};
+
+const updateEvent = async event => {
+    const {Id} = event;
+    return new Promise(((resolve, reject) => {
+        let data = [];
+        fs.createReadStream(filePath)
+            .on('error', (err) => {
+                reject(err)
+            })
+
+            .pipe(csv())
+            .on('data', (row) => {
+                if (Id.toString() === row.Id.toString()) {
+                    row = event;
+                }
+                data.push(row);
+            })
+
+            .on('end', () => {
+                jsonexport(data, (err, csv) => {
+                    if (err) return console.error(err);
+                    fs.writeFile(filePath, csv, {flag: 'w'}, (error => {
+                        console.log(error)
```

We would want to propagate an error further and not just stop here. In case of stopping the request will hang and timeout eventually.

Here we would probably want to `reject(error)` or reject with our custom error.